Upgrade react-native to 0.68.2

ofreyssinet-ledger opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already.
Disclaimer: Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

#### Disclaimer
Yes I know that RN 0.69.0 was released last week 😬 but I was already working on this and it's a 1st step forward that is needed to unblock us on some topics (see below). 

### 📝 Description

I'm doing this because we would like to be ready for [`react-native-video@6.0.x`](https://www.npmjs.com/package/react-native-video) ([which requires `react-native@0.68.2`](https://github.com/react-native-video/react-native-video/tree/691a11dc4ec4049739e05a35295b84dfc88ae466#a-video-component-for-react-native)) when it releases (right now it's in alpha) as it will solve a [dependency conflict with `storyly-react-native`](https://github.com/Netvent/storyly-mobile/issues/188) which we would like to use.

#### Main changes:
- Upgraded `react-native` to `0.68.2`, following [this guide](https://react-native-community.github.io/upgrade-helper/?from=0.67.3&to=0.68.2) and picked what works for us:
   - we don't upgrade Flipper as it crashes on runtime
   - we don't upgrade gradle as it builds fine like this and v7 didn't work out of the box
   - we don't keep `react-native-gradle-plugin` as it's only necessary for the new architecture..
   - we don't change `AppDelegate.m` to the new `AppDelegate.mm` as it's only useful for the new RN arch which we aren't using yet + it was a pain to migrate the existing config (Firebase, Flipper, splash screen)
- Upgraded `react-native-reanimated` to `2.8.0`
- Upgraded `lottie-react-native` to `5.1.3` as it was not building on iOS without upgrading -> I tested the device lotties in the "Debug Lottie" menu and it seems to work fine.
- Upgraded `react-native-gesture-handler` to `2.5.0` & [Migrating off RNGHEnabledRootView](https://docs.swmansion.com/react-native-gesture-handler/docs/guides/migrating-off-rnghenabledroot) as its setup on Android (in `MainActivity.java`) might conflict with react-native stuff later on
- Fixed an issue in the portfolio where if there was no assets, scrolling was crashing the app on iOS. This is a mysterious issue and the logs are similar to this issue https://github.com/software-mansion/react-native-reanimated/issues/2285, for now it has been solved by removing the animation of a border width (border which anyway was invisible so the animation was pointless).

#### What I tested:

I didn't test much of the app after fixing the obvious crashes & issues, just tested the very basic functionality of the app on real device for both platforms + things known to be susceptible to break while upgrading react-native (following list). **A full round of non-regression testing will have to be done on both platforms before merging**.

- [x]  Android (tested on OnePlus 8, staging build from [https://github.com/LedgerHQ/ledger-live/actions/runs/2591069426](https://github.com/LedgerHQ/ledger-live/actions/runs/2591069426))
    - [x]  QR code scanning → tested with an import account from LLD flow
    - [x]  Bluetooth pairing
    - [x]  Camera access → tested with an import account from LLD flow
    - [x]  Password lock
    - [x]  Fingerprint lock
    - [x]  dApp correct render → tested with Paraswap
    - [x]  Deeplinks -> tested with all deeplinks from testing page (`pnpm mobile test-deep-links`)

- [x]  iOS (tested on iPhone X, staging build from: [https://github.com/LedgerHQ/ledger-live-build/actions/runs/2623892281](https://github.com/LedgerHQ/ledger-live-build/actions/runs/2623892281))
  - [x]  QR code scanning → tested with an import account from LLD flow
  - [x]  Bluetooth pairing
  - [x]  Camera access → tested with an import account from LLD flow
  - [x]  Password lock
  - [x]  dApp correct render → tested with Paraswap
  - [x]  Deeplinks -> tested with all deeplinks from testing page (`pnpm mobile test-deep-links`)

#### TODO

- [x] fix Firebase features not found on iOS (broken config)
- [x] upgrade `react-native-gesture-handler` and [Migrating off RNGHEnabledRootView](https://docs.swmansion.com/react-native-gesture-handler/docs/guides/migrating-off-rnghenabledroot) as its setup on Android (in `MainActivity.java`) might conflict with react-native stuff later on. 
- [x] better documentation of this PR (document individual changes for easy code review)
- [x] changesets
- [x] fix CI builds:
  - [x] [Android: JDK 11 required](https://github.com/LedgerHQ/ledger-live/runs/7004931923?check_suite_focus=true)
  - [x] [Android: "Installed Build Tools revision 31.0.0 is corrupted"](https://github.com/LedgerHQ/ledger-live/runs/7109794357?check_suite_focus=true#step:17:1245)
  - [x] [iOS: pods](https://github.com/LedgerHQ/ledger-live/runs/7004878732?check_suite_focus=true)

Issues detected but not a regression (already happening on prev versions of RN):
- [x] iOS crash with react-native-reanimated in scrollView handleScroll when interpolating an animated value
  https://github.com/software-mansion/react-native-reanimated/issues/2285
  https://github.com/software-mansion/react-native-reanimated/issues/2764
  - [x] try applying patch from https://github.com/software-mansion/react-native-reanimated/pull/3194
     -> it makes the scroll a bit jittery so not ideal
- [ ] ⚠️⚠️ iOS crash on market detail page if scrolling up & down quickly (linked to the issue above & `ScrollContainerHeader`)

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` <!-- The list of end user projects impacted by the change. -->
- **Linked resource(s)**: [FAT-165] <!-- Attach any ticket number if relevant. (JIRA / Github issue number) -->

### ✅ Checklist

- [ ] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
- [x] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
- [x] **No breaking changes** <!-- If there are breaking changes, please explain why. -->

**A full round of non-regression testing will have to be done on both platforms before merging.**

### 📸 Demo

<!--
For visual features, please attach screenshots or video recordings to demonstrate the changes.
For libraries, you can add a code sample.
For bugfixes, you can drop this section.
-->

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

<!-- If any of the expectations are not met please explain the reason in detail. -->


[FAT-165]: https://ledgerhq.atlassian.net/browse/FAT-165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ofreyssinet-ledger commented on libs/ledger-live-common/mobile-test-app/android/.project:
```diff
@@ -14,4 +14,15 @@
 	<natures>
 		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
 	</natures>
+	<filteredResources>
```
tbh I don't know what this is and why it changed 😬

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/AndroidManifest.xml:
```diff
@@ -39,6 +39,7 @@
         android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
         android:windowSoftInputMode="adjustResize"
         android:screenOrientation="portrait"
+        android:exported="true"
```
[necessary for targeting API 31](https://stackoverflow.com/questions/70333565/targeting-s-version-31-and-above-requires-that-an-explicit-value-for-android)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/MainActivity.java:
```diff
@@ -111,25 +129,4 @@ protected void onResume() {
-        return new ReactActivityDelegateWrapper(this, new ReactActivityDelegate(this, getMainComponentName()) {
-            @Override
-            protected ReactRootView createRootView() {
-                return new RNGestureHandlerEnabledRootView(MainActivity.this);
```
See following section in PR description:
> - Upgraded `react-native-gesture-handler` to `2.5.0` & [Migrating off RNGHEnabledRootView](https://docs.swmansion.com/react-native-gesture-handler/docs/guides/migrating-off-rnghenabledroot) as its setup on Android (in `MainActivity.java`) might conflict with react-native stuff later on

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/Android.mk:
```diff
@@ -0,0 +1,40 @@
+THIS_DIR := $(call my-dir)
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/newarchitecture/modules/MainApplicationTurboModuleManagerDelegate.java:
```diff
@@ -0,0 +1,40 @@
+package com.ledger.live.newarchitecture.modules;
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/newarchitecture/components/MainComponentsRegistry.java:
```diff
@@ -0,0 +1,31 @@
+package com.ledger.live.newarchitecture.components;
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/MainApplicationModuleProvider.cpp:
```diff
@@ -0,0 +1,20 @@
+#include "MainApplicationModuleProvider.h"
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/MainApplicationModuleProvider.h:
```diff
@@ -0,0 +1,11 @@
+#pragma once
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/MainApplicationTurboModuleManagerDelegate.cpp:
```diff
@@ -0,0 +1,38 @@
+#include "MainApplicationTurboModuleManagerDelegate.h"
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/MainComponentsRegistry.cpp:
```diff
@@ -0,0 +1,48 @@
+#include "MainComponentsRegistry.h"
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/MainComponentsRegistry.h:
```diff
@@ -0,0 +1,24 @@
+#pragma once
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/app/src/main/jni/OnLoad.cpp:
```diff
@@ -0,0 +1,10 @@
+#include <fbjni/fbjni.h>
```
This file isn't used yet as we are not activating the new RN architecture (Fabric + TurboModules)

ofreyssinet-ledger commented on apps/ledger-live-mobile/android/gradlew:
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
taken from upgrade tool

ofreyssinet-ledger commented on apps/ledger-live-mobile/src/screens/Portfolio/Header.tsx:
```diff
@@ -96,38 +96,13 @@ function PortfolioHeader({
     };
   }, [graphCardEndPosition]);
 
-  const ContainerStyle = useAnimatedStyle(() => {
```
See following section of PR description:
> - Fixed an issue in the portfolio where if there was no assets, scrolling was crashing the app on iOS. This is a mysterious issue and the logs are similar to this issue https://github.com/software-mansion/react-native-reanimated/issues/2285, for now it has been solved by removing the animation of a border width (border which anyway was invisible so the animation was pointless).

gre reviewed (approved):
lgtm. big impact but necessary 🙏

Justkant reviewed (approved):
LGTM, tested on a real android device and the ios simulator

elbywan reviewed (approved):
LGTM! 👍 👌

gre commented:
very impressive work @ofreyssinet-ledger , i hope you manage to close all remaining topics 🤞

ofreyssinet-ledger commented:
@gre haha thanks! It's painful work but I think I'm finally done 🤞  at least with what QA detected in their last test:
- an issue with Cosmos send: that was because of a wrong conflict resolution of the pnpm lock file when I rebased countless times this PR on develop 😅 That caused the installation of a wrong version of the Cosmos SDK I believe.
- there was another issue with an animation in the RatingsModal on iOS, probably due to the upgrade of react-native-reanimated, I just fixed it.
